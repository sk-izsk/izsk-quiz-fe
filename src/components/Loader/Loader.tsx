import { makeStyles } from '@material-ui/core';
import React from 'react';
import { ProgressCircular } from 'ui-neumorphism';
import { CustomTheme, theme } from '../../theme/muiTheme';

export interface LoadingScreenProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  loaderContainer: {
    display: 'flex',
    width: '100vw',
    height: '100vh',
    justifyContent: 'center',
    alignItems: 'center',
  },
}));

const LoadingScreen: React.FC<LoadingScreenProps> = () => {
  const classes = useStyles();
  return (
    <div className={classes.loaderContainer}>
      <ProgressCircular indeterminate size={100} width={10} color={theme.palette.primary.main} />
    </div>
  );
};

export { LoadingScreen };
