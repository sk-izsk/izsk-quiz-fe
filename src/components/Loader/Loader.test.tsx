import { render } from '@testing-library/react';
import React from 'react';
import { LoadingScreen } from './Loader';

test('LoadingScreen renders correctly', () => {
  const { asFragment } = render(<LoadingScreen />);
  expect(asFragment()).toMatchSnapshot();
});
