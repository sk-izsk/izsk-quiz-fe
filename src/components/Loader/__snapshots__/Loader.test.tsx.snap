// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-loaderContainer-1"
  >
    <div
      aria-busy="true"
      class=""
    >
      <svg
        aria-label="audio-loading"
        height="200"
        stroke="#E63647"
        viewBox="0 0 44 44"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke-width="2"
        >
          <circle
            cx="22"
            cy="22"
            r="1"
          >
            <animate
              attributeName="r"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="0s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
          <circle
            cx="22"
            cy="22"
            r="1"
          >
            <animate
              attributeName="r"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.165, 0.84, 0.44, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 20"
            />
            <animate
              attributeName="strokeOpacity"
              begin="-0.9s"
              calcMode="spline"
              dur="1.8s"
              keySplines="0.3, 0.61, 0.355, 1"
              keyTimes="0; 1"
              repeatCount="indefinite"
              values="1; 0"
            />
          </circle>
        </g>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
